Eastern-Research-Group/mywaterway
# Feature/server 404 redirect

## Related Issues:
* [https://app.breeze.pm/projects/100762/cards/3345751](https://app.breeze.pm/projects/100762/cards/3345751)

## Main Changes:
*  Created a server side route to the 404.html page and updated the client side 404 error page so that it redirects to the server side version. This was done so that when users navigate to a page that doesn't exist, they will get a 404 http status code returned and the server side 404 error page. 
* Moved the `400.html` , `500.html` and `favicon.ico` files to the `server\public` folder so they are included in the build. Also, removed the `server\public_other_pages` folder. 

## Steps To Test:
1. Open the Chrome dev tools and go to the network tab
2. Navigate to [http://localhost:3000/thisdoesntexist](http://localhost:3000/thisdoesntexist)
3. Verify the page gets rerouted to [http://localhost:9090/400.html](http://localhost:9090/400.html)
    * The page should look the same as the old error page. 
    * When running locally the homepage link won't work, since it will try to route to [http://localhost:9090](http://localhost:9090) instead of [http://localhost:3000](http://localhost:3000). This won't be a problem on the develop/staging sites though. 
4. Verify that `404.html` request shows up in the network tab with a 404 status code.
